gstd/gstd_tcp.c

```diff
@@ -347,3 +381,25 @@ gstd_tcp_init_get_option_group (GstdIpc * base, GOptionGroup ** group)
   g_option_group_add_entries (*group, tcp_args);
   return TRUE;
 }
+
+static gboolean
+gstd_tcp_add_listeners(GSocketService *service, gchar * address, gint port, GError ** error)
+{
+  GSocketAddress * sa;
+  GError *err = NULL;
+  gboolean ret = TRUE;
+
+  sa = g_inet_socket_address_new_from_string(address, port);
+
+  if(g_socket_listener_add_address(G_SOCKET_LISTENER(service), sa,
+    G_SOCKET_TYPE_STREAM, G_SOCKET_PROTOCOL_TCP, NULL, NULL, &err)
+      == FALSE ){
+    g_printerr ("set_address: error while adding the address to socket listener: %s\n",
+	   err->message);
+    ret = FALSE;
```

Comment:
We need to free the GError here.

Reply:
GError is passed as argument into function. It is freed out of function.